src/raze/middleware/proxy.cr

```diff
@@ -0,0 +1,97 @@
+require "../handler"
+require "uri"
+
+# A middleware for proxying the request to another endpoint
+class Raze::Proxy < Raze::Handler
+  @target_hosts = [] of String
+  @target_path = ""
+  @next_target_host_index = 0
+  @headers : Hash(String, String) | Nil
+  @timeout : Int32 | Nil
+
+  def initialize(host : String, path = "", @lchop_proxy_path = "", @ignore_proxy_path = false, @headers = nil, @timeout = nil)
+    @target_hosts << host
+    @target_path = path
+    if timeout = @timeout
+      timeout = nil
+    end
+    validate_props
+  end
+
+  def initialize(host : Array(String), path = "", @lchop_proxy_path = "", @ignore_proxy_path = false, @headers = nil, @timeout = nil)
+    @target_hosts = host
```

Comment:
What about checking that array is not empty?

Reply:
good catch. thx